core/cmdbobject.class.inc.php

```diff
@@ -114,6 +114,31 @@ public static function SetCurrentChange($oChange)
 		self::$m_oCurrChange = $oChange;
 	}
 
+	/**
+	 * @param string $sUserInfo
+	 * @param string $sOrigin
+	 * @param \DateTime $oDate
+	 *
+	 * @throws \CoreException
+	 *
+	 * @since 2.7.4 3.0.0 N°3717 new method to reset current change
+	 */
+	public static function SetCurrentChangeFromParams($sUserInfo, $sOrigin = null, $oDate = null)
+	{
+		static::$m_oCurrChange = MetaModel::NewObject('CMDBChange');
+
+		static::$m_oCurrChange->Set("userinfo", $sUserInfo);
+
+		if (!is_null($sOrigin)) {
+			static::$m_oCurrChange->Set("origin", $sOrigin);
+		}
+
+		if (is_null($oDate)) {
+			$oDate = time();
+		}
+		static::$m_oCurrChange->Set("date", $oDate);
```

Comment:
What about
```suggestion
		static::SetTrackInfo($sUserInfo);
		static::SetTrackOrigin($sOrigin);
		static::CreateChange();

		if (!is_null($oDate)) {
			static::$m_oCurrChange->Set("date", $oDate);
		}
```

As it seems you are doing double code right now and it might be better to just reuse already existing code.

Reply:
I like this ! I'm committing this suggestion. Thanks !